tensorboard/functionaltests/wct_test_driver.py

```diff
@@ -47,6 +50,30 @@ def create_test_class(binary_path, web_path):
     the test file's main module.
   """
 
+  class BrowserLogIndicatesResult(object):
+    def __init__(self):
+      self.passed = False
+    def __call__(self, driver):
+      # Scan through the log entries and search for a text indicating whether
+      # the test passed or failed.
+      messages = [entry['message'] for entry in driver.get_log('browser')]
+      if self._log_contains(messages, 'failing test'):
+        self.passed = False
+        return True
+      if self._log_contains(messages, 'test suite passed'):
```

Comment:
Do you have a reference that indicates that the driver will only print
“test suite passed” once _all_ tests have passed? For instance, it’s not
obvious to me that this waits for all top-level `describe` blocks—I
checked, and it appears to, but I don’t see any guarantee of that.

Reply:
I'm not a 100% certain, but if I look at console.js in chrome's debugger there's a function called logSummary that seems to be called only at the end of the mocha run that logs 'test suite passed'.

Reply:
Okay, no problem—if we find out that this is insufficient then we can
update this test code again. :-)